engine-types/src/types/mod.rs

```diff
@@ -128,11 +131,14 @@ impl<T> Stack<T> {
         self.stack.push(value);
     }
 
+    #[must_use]
+    #[allow(clippy::missing_const_for_fn)]
```

Comment:
Why allow?

Reply:
And here seems a bug in the clippy. There a destructor of the `self` will be invoked which makes it impossible to make this method `const`.